subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/options/OptionReaderTest.groovy

```diff
@@ -631,6 +708,22 @@ class OptionReaderTest extends Specification {
         }
     }
 
+    public static class TestClassWithListPropertyField {
+        @Option(option = 'customOptionName', description = "custom description")
+        final ListProperty<String> field1
+
+        @Option(description = "Descr Field2")
+        final ListProperty<TestEnum> field2
```

Comment:
🤔 Instead of `field1` and `field2` how about using more descriptive names, like `stringList` and `enumList`?